### PR TITLE
Added the Introduction to the Contributor Journey Research

### PR DESCRIPTION
**Pull request by prajjwalyd**
# Changes

The introduction contains:
- Abstract
- Goals
- Objectives
- Key questions
- Hypotheses
- Methods

/kind documentation

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

**Review by Cali0707 — COMMENTED** [review]
@prajjwalyd this looks good to me! As discussed in the last UX WG meeting we will leave this open until the next UX WG meeting so others can review it, and we will approve as a group at the meeting.

Thanks for your work on this!

**Comment by Cali0707** [comment]
/cc @Leo6Leo @zainabhusain227 @mmejia02 @aliok 

Please leave comments! We'll do final approval and merge at the next WG meeting, but it would be great to request any changes before then

File: user-research/contributor-research/Introduction.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+## Research Goals:
+- Understand the current contributor journey within Knative.
+- Identify factors that contribute to contributors staying or leaving the Knative community.
+- Highlight areas of the contributor experience that can be improved to increase contributor retention and propose ways to improve those areas.
+
+## Objectives:
+- Gather insights into the onboarding process for new contributors to Knative.
+- Explore the experiences of both retained and departed contributors.
+- Identify pain points and areas of dissatisfaction throughout the contributor journey.
+- Investigate the effectiveness of existing resources and documentation for contributors.
```
Comment (Leo6Leo) [inline user-research/contributor-research/Introduction.md:14]: Just a thought: Explore with Existing Literature

Briefly mentioning existing research or literature on open-source contributor engagement and retention could be helpful. Maybe there are some good research or good practice happening in other open source projects.

File: user-research/contributor-research/Introduction.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+- Are there any differences in experiences between contributors with varying levels of technical expertise or backgrounds?
+- What is missing in the current contributor experience of Knative as compared to some other well-established open-source projects?
+
+## Hypotheses:
+- Contributors who have a positive onboarding experience, are proactive, and receive clear guidance and support, are more likely to stay engaged with the Knative community.
+- Factors such as perceived value of contributions, prospective future opportunities, and alignment with personal or professional goals influence contributors' decisions to remain active in the project.
+- Contributors may disengage due to challenges like unclear contribution processes, technical knowledge/expertise barriers, insufficient/delayed feedback, lack of recognition, community integration issues, etc. Additionally, heavy workloads, limited free time, personal commitments, or simply a change of plans may also contribute to disengagement.
+
+
```
Comment (Leo6Leo) [inline user-research/contributor-research/Introduction.md:31]: Consider adding a section on the potential impact of your research findings and how you envision the implementation of your recommendations within the Knative community could be cool? This not only shows the practical relevance of your research but also can help in convincing the community.

**Review by Leo6Leo — COMMENTED** [review]
Just left some my thoughts, the intro is well-structured, great start @prajjwalyd !!

File: user-research/contributor-research/Introduction.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+## Abstract:
+Knative is a vital tool in the open-source and cloud-native ecosystem, enabling the development of enterprise-level serverless and event-driven applications. However, its complexity poses challenges for both new contributors and maintainers. Providing guidance consumes valuable maintainer time, resulting in challenges when contributors leave after limited engagement.
```
Comment (Cali0707) [inline user-research/contributor-research/Introduction.md:2]: I'm not totally sure about the last two sentences on this line. They seem to state your hypotheses as facts, when we are actually seeking to understand what the challenges are.

File: user-research/contributor-research/Introduction.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+This research aims to understand why some individuals sustain active involvement in Knative while others contribute sporadically or disengage entirely. The ultimate goal is to identify aspects of the contributor experience that could be improved to enhance contributor retention and propose actionable recommendations to tackle these issues.
+
+## Research Goals:
+- Understand the current contributor journey within Knative.
+- Identify factors that contribute to contributors staying or leaving the Knative community.
+- Highlight areas of the contributor experience that can be improved to increase contributor retention and propose ways to improve those areas.
+
+## Objectives:
```
Comment (Cali0707) [inline user-research/contributor-research/Introduction.md:10]: What is the difference between research goals and objectives?

**Review by Cali0707 — COMMENTED** [review]
This is a great start @prajjwalyd - awesome work! I just left a few comments where I was confused or unsure of a few sentences.

File: user-research/contributor-research/Introduction.md (hunk 1 above)
Comment (prajjwalyd) [reply user-research/contributor-research/Introduction.md:13]: Just added a whole new 'Methods' section from my drafts!

File: user-research/contributor-research/Introduction.md (hunk 2 above)
Comment (prajjwalyd) [reply user-research/contributor-research/Introduction.md:42]: covered this point in the 'Abstract' section.

File: user-research/contributor-research/Introduction.md (hunk 3 above)
Comment (prajjwalyd) [reply user-research/contributor-research/Introduction.md]: just made some changes to this part. please review.

File: user-research/contributor-research/Introduction.md (hunk 4 above)
Comment (prajjwalyd) [reply user-research/contributor-research/Introduction.md]: Goals are like the bigger picture, while objectives are the smaller tasks to make those goals happen... 
For eg. to achieve the goal of understanding the current contributor journey within Knative (1st goal), there are a few potential objectives (1, 2 and to some extent, the 3rd one too).

**Review by Cali0707 — APPROVED** [review]
/lgtm
/approve

Thanks @prajjwalyd - great work!

**Pull request merged**